### PR TITLE
Plugins | Filedelete | Add hook to `ZwCreateSection()'

### PR DESCRIPTION
**Pull request by skvl**
Some applications use sections and views to read/write files.

Add handles of files for which section with `SECTION_MAP_WRITE` flag have been
created into registry of changed files.
Filter out system handles. Those having mask `0xffffffff80000000`.

### CODE REVIEW

**Comment by skvl** [comment]
@tklengyel unfortunately I don't see results yet. Though I have found *astyle* bug and fixed it already.

**Comment by tklengyel** [comment]
```
*** CID 190892:  Null pointer dereferences  (REVERSE_INULL)
/src/plugins/filedelete/filedelete2.cpp: 560 in start_readfile(drakvuf *, drakvuf_trap_info *, vmi_instance *, unsigned long)()
554         return response;
555     
556     err:
557         if (restore_regs)
558             memcpy(info->regs, &injector->saved_regs, sizeof(x86_registers_t));
559     
>>>     CID 190892:  Null pointer dereferences  (REVERSE_INULL)
>>>     Null-checking "injector" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.
560         if (injector)
561             delete injector;
562     
563         return response;
564     }
565     
```

```
*** CID 190891:    (NULL_RETURNS)
/src/plugins/filedelete/filedelete.cpp: 252 in extract_ca_file(filedelete *, drakvuf *, const drakvuf_trap_info *, vmi_instance *, unsigned long, access_context_t *, const _ustring *)()
246                 {
247                     uint8_t page[4096];
248     
249                     if ( VMI_FAILURE == vmi_read_pa(vmi, VMI_BIT_MASK(12,48) & pte, 4096, &page, NULL) )
250                         continue;
251     
>>>     CID 190891:    (NULL_RETURNS)
>>>     Dereferencing a pointer that might be null "fp" when calling "fseek".
252                     if ( !fseek ( fp, fileoffset, SEEK_SET ) )
253                         fwrite(page, 4096, 1, fp);
254                 }
255             }
256     
257             ctx->addr = subsection + f->offsets[SUBSECTION_NEXTSUBSECTION];
/src/plugins/filedelete/filedelete.cpp: 262 in extract_ca_file(filedelete *, drakvuf *, const drakvuf_trap_info *, vmi_instance *, unsigned long, access_context_t *, const _ustring *)()
256     
257             ctx->addr = subsection + f->offsets[SUBSECTION_NEXTSUBSECTION];
258             if ( !vmi_read_addr(vmi, ctx, &subsection) )
259                 break;
260         }
261     
>>>     CID 190891:    (NULL_RETURNS)
>>>     Dereferencing a pointer that might be null "fp" when calling "fclose".
262         fclose(fp);
263         free(file);
264     
265         save_file_metadata(f, info, curr_sequence_number, control_area, filename);
266     }
267

**Comment by skvl** [comment]
@tklengyel I have fixed first warning. But the second is not in my code. I didn't even touch it.

NB It's funny but first warning was introduced in #404 but there were no warnings. Even now I don't see any warnings from Travic-CI. The whole thisg is "green".

**Comment by tklengyel** [comment]
Coverity only runs on merged PRs, not when it is opened. When it detects errors it doesn't block the PR through Travis.

**Comment by skvl** [comment]
Is this OK now?

**Comment by tklengyel** [comment]
Not sure why Coverity detected that error only after your PR was merged, but if you are at it, might as well fix it now :)

File: src/plugins/filedelete/filedelete.cpp [hunk 1]
```diff
@@ -201,6 +201,9 @@ static void extract_ca_file(filedelete* f,
 
     FILE* fp = fopen(file, "w");
 
+    if (!fp)
+        return;
```
Comment (tklengyel) [inline src/plugins/filedelete/filedelete.cpp]: Yeap, this should do it.

File: src/plugins/filedelete/filedelete2.cpp [hunk 2]
```diff
@@ -714,6 +713,57 @@ static event_response_t writefile_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* inf
...
+
+    if (f->pm == VMI_PM_IA32E)
+    {
+        access_mask = (uint32_t) info->regs->rdx;
+        ctx.addr = info->regs->rsp + 7 * sizeof(uint64_t);
+        if ( VMI_FAILURE == vmi_read_64(vmi, &ctx, (uint64_t*) &handle) )
+            goto done;
+        // Filter out system handles: those having high bits rised
+        // WARNING Without this target VM could freeze or crash!
+        if (handle & ~(0x7fffffffULL))
```
Comment (tklengyel) [inline src/plugins/filedelete/filedelete2.cpp]: So this is easier to filter by casting handle to be a signed integer and then just checking if "handle < 0"

Comment (skvl) [reply src/plugins/filedelete/filedelete2.cpp]: This is sort of mask. I think it more clear what I check here.

Comment (tklengyel) [reply src/plugins/filedelete/filedelete2.cpp]: No, I think it's more clear what you check for here if its a negative integer check and not the inverse of a magic hex value. That's effectively what this is.

Comment (skvl) [reply src/plugins/filedelete/filedelete2.cpp]: @tklengyel, Done!

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test.

**Comment by tklengyel** [comment]
It worked: `        Defects eliminated: 2`

**Pull request merged**